src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -662,19 +668,32 @@ func (s *commitLogSource) readCommitLog(namespaces bootstrap.Namespaces, span op
 			continue
 		}
 
+		arg := accumulateArg{
+			namespace: seriesEntry.namespace,
+			series:    seriesEntry.series,
+			shard:     seriesEntry.series.Shard,
+			dp:        entry.Datapoint,
+			unit:      entry.Unit,
+		}
+
+		annotationLen := len(entry.Annotation)
+		if annotationLen > 0 {
+			// Use the predefined buffer if the annotation fits in it.
+			if annotationLen <= len(arg.shortAnnotation) {
+				copy(arg.shortAnnotation[:], entry.Annotation)
```

Comment:
How does this short annotation buffer help you here since you're creating a new accumulateArg each time through? Wouldn't you need to reuse this arg for a buffer to be beneficial?

Reply:
A good question. The main goal here is to avoid heap allocation. The struct (together with the fixed size array that is embedded in it) gets created on the stack, then copied to the channel, and then copied out of the channel. It never escapes to heap.

Reply:
Ahh, clever. Sgtm.

Reply:
Oh interesting. Even with the `namespace *bootstrapNamespace` the accumulateArg gets created on the stack? Wouldn't have expected that. Thought since it had a pointer and was being sent over a channel it would escape.

Reply:
The pointers that are in the struct and point to some external object are not a problem, they simply get copied as part of the struct.
The problem would be external pointers to this struct which cause it escape to the heap. Also pointers (slice references) pointing to that fixed array which is an embedded part of the struct.